Nodejs/Setup/InstallerTelemetryAction/InstallerTelemetryAction.cs

```diff
@@ -0,0 +1,124 @@
+﻿//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Deployment.WindowsInstaller;
+using System.Diagnostics;
+using System.Net;
+using System.Collections.Specialized;
+using System.Web.Script.Serialization;
+using System.Globalization;
+
+namespace Microsoft.NodejsTools
+{
+    public class InstallerTelemetryActions {
+        private static void LogInstallStatus(string InstallStatus,Session session) {
+            TimeSpan installTime = DateTime.Now - DateTime.Parse(session["InstallStartTime"]);
+            bool isInstalled = session.EvaluateCondition("Installed");
+            string currentState = null;
+            string requestState = null;
+
+            FeatureInfoCollection featureInfoCollection = session.Features;
+            foreach (FeatureInfo featureInfo in featureInfoCollection) {
+                currentState = featureInfo.CurrentState.ToString();
+                requestState = featureInfo.RequestState.ToString();

```

Comment:
Please use InvariantCulture everywhere since none of these operations should be culture dependent.